### ISSUE TITLE
Switch to official Logstash Docker image

### ISSUE THREAD

**Issue by loshkovskyi** [CiscoCloud/microservices-infrastructure#596]
As far as I remember the only reason we built our own Logstash Docker image was Kafka support (added in Logstash 1.5). Since the official Logstash 1.5.3 has been already released, it might be a good idea to start using it and deprecate our own image. Please let me know if there are any objections.

**Comment by tgermain** [comment]
Definetly a +1. 

My only concern is for container logs forwarding : 
we mount docker directory in logstash container ` --volume=/var/lib/docker/containers:/logs/docker:ro `. I remember having permissions issue to read the logs files. 

You may need change the permissions of `/logs/docker` at runtime.

**Comment by loshkovskyi** [comment]
Hello @tgermain . I have prepared the new Logstash image. How can I reproduce the permissions issue when reading the log files? Please also provide some details on your environment, for example, if you deployed M-I on OpenStack or GCE. Thank you.

**Comment by tgermain** [comment]
Hi @loshkovskyi, 


I've adapt M-I to deploy it on bare metal but the log & metrics part is intact (collectd & syslog & mesos logs => local logstash in a container).


# the permission issue :
if you launch 
```
docker run -ti \
--log-driver=none --rm \
-v /var/lib/docker/containers:/logs/docker:ro \
-v /var/run/docker.sock:/var/run/docker.sock:ro \
logstash:1.5.3 agent -e "input {file { path => '/logs/docker/*/*-json.log' type => docker codec => json}} filter {} output{stdout { codec => rubydebug }}"
```
which is a dummy inline configuration which read docker logs and write it to stdout. 
It does not show the log from another running container like `docker run ubuntu echo"hello"`

I know why : the official image use `gosu` to launch logstash. It's a good practice not run process as root in container but in out case it's problematic. 
I made a version of logstash based on the official which run as root, it's working well.

From there you have 2 choices :
- run logstash as root which is a bad idea IMHO
- mount `/var/lib/docker/containers` not as read only and change permissions before running logstash.

**Comment by loshkovskyi** [comment]
Thank you @tgermain , I was able to reproduce this issue. There is ongoing discussion here: https://github.com/docker/docker/issues/7198 related to this problem. In my opinion, we should `chown` the Docker logs by extending the entrypoint script `docker-entrypoint.sh`. I tested this solution it seems to work. I'm currently discussing some things concerning the Logstash image with other team members, that is why I will postpone my pull request till next week.
